### DISCUSSION TITLE
Will we have a schedule for release version?

### DISCUSSION THREAD

**Discussion by IlysvlVEizbr** [micropython/micropython#10734] [General]
One idea is that: the schedule specifies a release cycle (e.g., about a few months), and no new features or new ports support will be merged to the main branch in the last few weeks of the cycle, only bug fixes.

So that, some users can experience some of the new features in a more stable (supposed) environment. And some micropython related repositories (e.g., the [One-click installation version of mpy-cross](https://pypi.org/project/mpy-cross/)) would be able to confirm when to release updates to catch up with the latest changes to micropython and not become obsolete.

**Comment by jimmo** [comment]
> One idea is that: the schedule specifies a release cycle (e.g., about a few months), and no new features or new ports support will be merged to the main branch in the last few weeks of the cycle, only bug fixes.

Yes this is exactly the plan and we were working on a release-every-two-months plan, but several factors have made it difficult to put out the most recent release. 1.20 is coming very soon and we intend to go back to the regular releases like in the first half of 2022.

> So that, some users can experience some of the new features in a more stable (supposed) environment.

We aim for the nighties to be very stable. New ports don't affect other ports, etc.

> And some micropython related repositories (e.g., the [One-click installation version of mpy-cross](https://pypi.org/project/mpy-cross/)) would be able to confirm when to release updates to catch up with the latest changes to micropython and not become obsolete.

I'm not quite sure what you mean here. One note is that the mpy-cross package you link to is actually maintained by a third party, but we plan to take ownership of this ASAP and make it part of the regular release process. However the current version does track the micropython releases closely.

**Reply by IlysvlVEizbr** [reply]
> I'm not quite sure what you mean here. One note is that the mpy-cross package you link to is actually maintained by a third party,

  Yes, I mean that third party packages like this (at least for now), they would tend to release new versions after mpy releases. So if mpy only have nighties updates over a period of time, they would not likely to catch up with mpy.
  I have noticed that the sub-version of this third party mpy-cross is not the same as the latest mpy nighties, but it has not been updated to ensure that mpy-cross will still work with the release version of mpy.

> Yes this is exactly the plan and we were working on a release-every-two-months plan,

  But the problem I mentioned above would be much better if mpy decided to update the release every 2 months or so.

**Reply by Josverl** [reply]
@jimmo I hope&assume that mpremote will follow the same cadence, as it's just impossible to install that from GitHub without a local clone currently

**Comment by ma261065** [comment]
> 1.20 is coming very soon 

Are there any estimated timings around the 1.20 release other than "very soon" :-)
